### ISSUE TITLE
🐛 Panic running biome lint . and piping to jq

### ISSUE THREAD

**Issue by dmmulroy** [biomejs/biome#3469]
### Environment information

```block
CLI:
  Version:                      1.7.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v16.20.2"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/8.15.7"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

I attempted to run the following command(s) on one of our monorepos at Vercel:

```sh
󰘧 pn biome lint --log-kind=compact --reporter=json --max-diagnostics=50000 . | jq '.diagnostics | group_by(.category) | map({
      category: .[0].category,
      count: length,
      filePaths: map(.location.path.file) | unique
    }) | sort_by(.count) | reverse' > biome_report.json

```

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates/biome_console/src/lib.rs:148:14
Thread Name: main
Message: called `Result::unwrap()` on an `Err` value: Os { code: 32, kind: BrokenPipe, message: "Broken pipe" }

### Expected result

It should not throw an error

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
Could you try to find the file / code that resulted in this crash? Just to be sure: does Biome failed without jq and `--reporter`?

**Comment by Conaclos** [comment]
The error seems coming from Biome trying to use terminal colors.
Could you try with the option `--colors=off` (and `--colors=force`)?

**Comment by ematipico** [comment]
I am going to close this, since we haven't had an answer for 2 weeks.

Also, this was fixed in `main`, where we disable colours when using `--reporter`

**Comment by dmmulroy** [comment]
Sorry - totally lost track of this, I was never able to reproduce it again and we've successfully migrated 1 or 2 giant monorepos to biome 💜


https://x.com/anthonysheww/status/1820885098134434237
